webSites/webSites-war/src/main/java/org/silverpeas/components/websites/servlets/WebSitesRequestRouter.java

```diff
@@ -90,1056 +95,1093 @@ public WebSiteSessionController createComponentSessionController(
+        scc.addPublicationToFolder(pubId, idTopic);
+        publish = true;
+      }
+    }
 
-        scc.notifyPublishers(auteur, nomSite, description, nomPage, listeMessage, date);
+    if (publish) {
+      arrayToClassify.add(site.getId());
+      scc.publish(arrayToClassify); // set etat du site a 1
+    }
+  }
 
-        request.setAttribute("SuggestionName", nomSite);
-        request.setAttribute("SuggestionUrl", nomPage);
-        destination = getDestination("Main", scc, request);
-      } else if ("DisplaySite".equals(function)) {
-        destination = request.getParameter("SitePage");
-      } else if (function.startsWith("ToWysiwyg")) {
-        String path = request.getParameter("path");
-        scc.checkPath(path);
-        String name = request.getParameter("name");
-        String nameSite = request.getParameter("nameSite");
-        String id = request.getParameter("id");
-
-        destination = "/wysiwyg/jsp/htmlEditor.jsp?";
-        destination += "SpaceId=" + scc.getSpaceId();
-
-        destination += "&SpaceName=" + URLEncoder.encode(scc.getSpaceLabel(), "UTF-8");
-        destination += "&ComponentId=" + scc.getComponentId();
-        destination += "&ComponentName=" + URLEncoder.encode(scc.getComponentLabel(), "UTF-8");
-        destination += "&BrowseInfo=" + URLEncoder.encode(nameSite, "UTF-8");
-        destination += "&Language=fr";
-        destination += "&ObjectId=" + id;
-        destination += "&FileName=" + URLEncoder.encode(name, "UTF-8");
-        destination += "&Path=" + URLEncoder.encode(path, "UTF-8");
-        destination += "&ReturnUrl=" + URLEncoder.encode(URLUtil.getApplicationURL() +
-            URLUtil.getURL(scc.getSpaceId(), scc.getComponentId()) + "FromWysiwyg?path=" + path +
-            "&name=" + name + "&nameSite=" + nameSite + "&profile=" + flag + "&id=" + id, "UTF-8");
+  private String uploadFile(final WebSiteSessionController scc, final HttpRequest request)
+      throws Exception {
 
-      } else if (function.startsWith("FromWysiwyg")) {
-        String path = request.getParameter("path");
-        scc.checkPath(path);
-        String id = request.getParameter("id");
+    List<FileItem> items = request.getFileItems();
+    boolean status = true;
+    String thePath = FileUploadUtil.getParameter(items, "path");
+    scc.checkPath(thePath);
+    FileItem item = FileUploadUtil.getFile(items);
+    if (item != null) {
+      status = (scc.addFileIntoWebSite(thePath, item) == 0);
+    }
 
-        SiteDetail site = scc.getWebSite(id);
-        request.setAttribute("Site", site);
+    request.setAttribute("UploadOk", status);
 
-        destination = "/webSites/jsp/design.jsp?Action=design&path=" + path + "&Id=" + id;
+    return  "/webSites/jsp/uploadFile.jsp?path=" + thePath;
+  }
 
-      } else if ("TopicUp".equals(function)) {
-        String topicId = request.getParameter("Id");
+  private String designSite(final WebSiteSessionController scc, final HttpRequest request)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "designSite" is 12 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)